crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -463,8 +463,8 @@ impl CumulocityConverter {
         let mut software_update_request = update_software
             .from_smartrest(smartrest)?
             .to_thin_edge_json()?;
-
-        let token = self.http_proxy.get_jwt_token().await?;
+        // get a fresh token not the cached one
+        let token = self.http_proxy.get_fresh_jwt_token().await?;
```

Comment:
This comment is useless as the method is well named.

If you put a comment, tell why. e.g "Make sure the agent will be given a fresh token as it cannot request a new one"